pkg/result/processors/fixer.go

```diff
@@ -43,218 +53,221 @@ func (p Fixer) Process(issues []result.Issue) ([]result.Issue, error) {
 	}
 
 	p.printStat()
 
-	return outIssues, nil
+	return notFixableIssues, nil
 }
 
-func (Fixer) Finish() {}
+//nolint:funlen,gocyclo // This function should not be split.
+func (p Fixer) process(issues []result.Issue) ([]result.Issue, error) {
+	// filenames / linters / edits
+	editsByLinter := make(map[string]map[string][]diff.Edit)
 
-func (p Fixer) fixIssuesInFile(filePath string, issues []result.Issue) error {
-	// TODO: don't read the whole file into memory: read line by line;
-	// can't just use bufio.scanner: it has a line length limit
-	origFileData, err := p.fileCache.GetFileBytes(filePath)
-	if err != nil {
-		return fmt.Errorf("failed to get file bytes for %s: %w", filePath, err)
-	}
+	var notFixableIssues []result.Issue
 
-	origFileLines := bytes.Split(origFileData, []byte("\n"))
+	for i := range issues {
+		issue := issues[i]
 
-	tmpFileName := filepath.Join(filepath.Dir(filePath), fmt.Sprintf(".%s.golangci_fix", filepath.Base(filePath)))
+		if issue.SuggestedFixes == nil || skipTextEditWithoutPosition(&issue) {
+			notFixableIssues = append(notFixableIssues, issue)
+			continue
+		}
 
-	tmpOutFile, err := os.Create(tmpFileName)
-	if err != nil {
-		return fmt.Errorf("failed to make file %s: %w", tmpFileName, err)
+		for _, sf := range issue.SuggestedFixes {
+			for _, edit := range sf.TextEdits {
+				start, end := edit.Pos, edit.End
+				if start > end {
+					return nil, fmt.Errorf("%q suggests invalid fix: pos (%v) > end (%v)",
+						issue.FromLinter, edit.Pos, edit.End)
+				}
+
+				edit := diff.Edit{
+					Start: int(start),
+					End:   int(end),
+					New:   string(edit.NewText),
+				}
+
+				if _, ok := editsByLinter[issue.FilePath()]; !ok {
+					editsByLinter[issue.FilePath()] = make(map[string][]diff.Edit)
+				}
+
+				editsByLinter[issue.FilePath()][issue.FromLinter] = append(editsByLinter[issue.FilePath()][issue.FromLinter], edit)
+			}
+		}
 	}
 
-	// merge multiple issues per line into one issue
-	issuesPerLine := map[int][]result.Issue{}
-	for i := range issues {
-		issue := &issues[i]
-		issuesPerLine[issue.Line()] = append(issuesPerLine[issue.Line()], *issue)
-	}
+	// Validate and group the edits to each actual file.
+	editsByPath := make(map[string][]diff.Edit)
+	for path, linterToEdits := range editsByLinter {
+		excludedLinters := make(map[string]struct{})
 
-	issues = issues[:0] // reuse the same memory
-	for line, lineIssues := range issuesPerLine {
-		if mergedIssue := p.mergeLineIssues(line, lineIssues, origFileLines); mergedIssue != nil {
-			issues = append(issues, *mergedIssue)
+		linters := maps.Keys(linterToEdits)
+
+		// Does any linter create conflicting edits?
+		for _, linter := range linters {
+			edits := linterToEdits[linter]
+			if _, invalid := validateEdits(edits); invalid > 0 {
+				name, x, y := linter, edits[invalid-1], edits[invalid]
+				excludedLinters[name] = struct{}{}
+
+				err := diff3Conflict(path, name, name, []diff.Edit{x}, []diff.Edit{y})
+				// TODO(ldez) TUI?
```

Comment:
Terminal User Interface? 🤔 

This is great with a stable order, applying the one fix and warning the user. Best case scenario, it resolves all errors. Worst case, the user might have to apply fixes again.

Reply:
Yes this Terminal User Interface.
I put this TODO here because it's when a linter suggests several changes for one line.

> This is great with a stable order, applying the one fix and warning the user. 

Cleary is not a good idea.

ex: with `!!a` the suggestions will be: `!a` or `a`. We cannot choose one automatically.